src/libraries/Common/tests/System/Security/Cryptography/MLKemTests.Keys.cs

```diff
@@ -221,6 +239,58 @@ public static void ExportDecapsulationKey_OnlyHasEncapsulationKey()
                 new byte[vector.Algorithm.DecapsulationKeySizeInBytes]));
         }
 
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void SubjectPublicKeyInfo_MLKem512_Ietf(bool tryExport)
+        {
+            using MLKem kem = MLKem.ImportPrivateSeed(MLKemAlgorithm.MLKem512, IncrementalSeed);
+            AssertSubjectPublicKeyInfo(kem, tryExport, MLKemTestData.IetfMlKem512Spki);
+        }
+
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void SubjectPublicKeyInfo_MLKem768_Ietf(bool tryExport)
+        {
+            using MLKem kem = MLKem.ImportPrivateSeed(MLKemAlgorithm.MLKem768, IncrementalSeed);
+            AssertSubjectPublicKeyInfo(kem, tryExport, MLKemTestData.IetfMlKem768Spki);
+        }
+
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void SubjectPublicKeyInfo_MLKem1024_Ietf(bool tryExport)
+        {
+            using MLKem kem = MLKem.ImportPrivateSeed(MLKemAlgorithm.MLKem1024, IncrementalSeed);
+            AssertSubjectPublicKeyInfo(kem, tryExport, MLKemTestData.IetfMlKem1024Spki);
+        }
+
+        private static void AssertSubjectPublicKeyInfo(MLKem kem, bool tryExport, ReadOnlySpan<byte> expectedSpki)
+        {
+            byte[] spki;
+            int written;
+
+            if (tryExport)
+            {
+                spki = new byte[kem.Algorithm.EncapsulationKeySizeInBytes + 22]; // 22 bytes of ASN.1 overhead.
+                Assert.True(kem.TryExportSubjectPublicKeyInfo(spki, out written), nameof(kem.TryExportSubjectPublicKeyInfo));
```

Comment:
No goldilocks tests?  (Might make sense as a dedicated thing, vs in the helper, since it's more about "the base class" than "the instance")